lib/matplotlib/axes/_secondary_axes.py

```diff
@@ -100,19 +100,13 @@ def set_alignment(self, align):
             either 'top' or 'bottom' for orientation='x' or
             'left' or 'right' for orientation='y' axis.
         """
-        if align in self._locstrings:
-            if align == self._locstrings[1]:
-                # need to change the orientation.
-                self._locstrings = self._locstrings[::-1]
-            elif align != self._locstrings[0]:
-                raise ValueError('"{}" is not a valid axis orientation, '
```

Comment:
the logic here was incorrect (we already checked that `align in self._locstrings`).